src/opm/input/eclipse/Schedule/Well/WellConnections.cpp

```diff
@@ -441,6 +508,228 @@ namespace Opm {
+        external::cvf::Vec3d p_bot(coord_bot[0], coord_bot[1], coord_bot[2]);
+        std::vector<external::cvf::Vec3d> points{p_top, p_bot};
+        std::vector<double> md_interval{perf_top.getSIDouble(0), perf_bot.getSIDouble(0)};
+        
+        external::cvf::ref<external::RigWellPath> wellPathGeometry = new external::RigWellPath;
+        wellPathGeometry->setWellPathPoints(points);
+        wellPathGeometry->setMeasuredDepths(md_interval);
+        external::cvf::ref<external::RigEclipseWellLogExtractor> e = new external::RigEclipseWellLogExtractor(wellPathGeometry.p(), *ecl_grid, cellSearchTree);
+        
+        // Keep the AABB search tree of the grid  to avoid redoing an expensive calulation 
+        cellSearchTree = e->getCellSearchTree();
+
+        // This gives the intersected grid cells IJK, cell face entrance & exit cell face point and connection length 
+        auto intersections = e->cellIntersectionInfosAlongWellPath();
+
+        int I{0};
+        int J{0};
+        int k{0};
+        for (size_t is = 0; is < intersections.size(); ++is){
+            auto ijk = std::array<int, 3>{};
+            ijk = ecl_grid->getIJK(intersections[is].globCellIndex);
+            I = ijk[0];
+            J = ijk[1];
+            k = ijk[2];
+            // std::cout<< "I: " << I << " J: " << J << " K: " << k << std::endl;
+            external::cvf::Vec3d connection_vector = intersections[is].intersectionLengthsInCellCS;
+
+
+            const CompletedCells::Cell& cell = grid.get_cell(I, J, k);
+
+           if (!cell.is_active()) {
+                auto msg = fmt::format("Problem with COMPTRAJ keyword\n"
+                                       "In {} line {}\n"
+                                       "The cell ({},{},{}) in well {} is not active and the connection will be ignored", location.filename, location.lineno, I,J,k, wname);
+                OpmLog::warning(msg);
+                continue;
+            }
+            const auto& props = cell.props;
+            double CF = -1;
+            double Kh = -1;
+            double r0 = -1;
+            auto ctf_kind = ::Opm::Connection::CTFKind::DeckValue;
+
+            if (defaultSatTable)
+                satTableId = props->satnum;
+
+            auto same_ijk = [&]( const Connection& c ) {
+                return c.sameCoordinate( I,J,k );
+            };
+
+            // if (r0Item.hasValue(0))
+            //     r0 = r0Item.getSIDouble(0);
+
+            if (KhItem.hasValue(0) && KhItem.getSIDouble(0) > 0.0)
+                Kh = KhItem.getSIDouble(0);
+
+            if (CFItem.hasValue(0) && CFItem.getSIDouble(0) > 0.0)
+                CF = CFItem.getSIDouble(0);
+
+            std::array<double,3> cell_size = cell.dimensions;
+
+            /* We start with the absolute happy path; both CF and Kh are explicitly given in the deck. */
+            if (CF > 0 && Kh > 0)
+                goto CF_done;
```

Comment:
Would it be possible to not use goto, but
```
if (CF < 0 && Kh < 0) {
...
}
else{
  if (! (CF > 0 && Kh > 0) ){
  ....
}}

Reply:
That is more neat indeed. I followed your suggestion.